spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -855,19 +855,57 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           None
 
         case EqualTo(left, right) =>
-          val leftExpr = exprToProtoInternal(left, inputs)
-          val rightExpr = exprToProtoInternal(right, inputs)
+          val zero = Literal.default(left.dataType)
+          val negZero = UnaryMinus(zero)
 
-          if (leftExpr.isDefined && rightExpr.isDefined) {
-            val builder = ExprOuterClass.Equal.newBuilder()
-            builder.setLeft(leftExpr.get)
-            builder.setRight(rightExpr.get)
+          def buildEqualExpr(leftExpr: Expr, rightExpr: Expr): ExprOuterClass.Expr = {
```

Comment:
Could you add a comment to mention this is a workaround until https://github.com/apache/datafusion/issues/11108 is resolved?